Extension/src/LanguageServer/configurations.ts

```diff
@@ -764,6 +781,10 @@ export class CppProperties {
                 configuration.includePath = includePath.concat(this.nodeAddonIncludes.filter(i => includePath.indexOf(i) < 0));
             }
             configuration.defines = this.updateConfigurationStringArray(configuration.defines, settings.defaultDefines, env);
+
+            configuration.dotconfig = this.updateConfigurationString(configuration.dotconfig, settings.defaultDotconfig, env);
+            configuration.defines = configuration.defines?.concat(this.getDotconfigDefines(configuration.dotconfig as string));
```

Comment:
Can configuration.defines be null or undefined? If so, I don't think this uses the .config defines.

Reply:
hmm, yeah the linter complained if I don't use the `?` here, you are right, let me check this